### PR TITLE
Move WASM-VM legacy branches into separate repos

### PR DESCRIPTION
**Pull request by camilbancioiu**
The WASM-VM repo contains two branches, `feat/arwen-v1_2` and `feat/arwen-v1_3`, which are kept as legacy versions of the VM. For better code management, these two branches have been transferred to standalone repositories: `wasm-vm-v1_2` and `wasm-vm-v1_3`.

This change obviates the need for `replace` statements in the `go.mod` file of `elrond-go`, targeting the separate versions of the VM.

The existing `wasm-vm` repo remains the current working VM. Backporting any changes, for backwards compatibility, must now be done on the aforementioned repos.

In this PR, the `go.mod` file in `elrond-go` references the `enable-epoch-handler` branches in all three versions of the VM. These branches have been updated to match the new repo structure.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -50,8 +50,4 @@ require (
 
 replace github.com/gogo/protobuf => github.com/ElrondNetwork/protobuf v1.3.2
 
-replace github.com/ElrondNetwork/arwen-wasm-vm/v1_2 v1.2.42-0.20220729115258-b9f2fb2f6568 => github.com/ElrondNetwork/arwen-wasm-vm v1.2.42-0.20220729115258-b9f2fb2f6568
-
-replace github.com/ElrondNetwork/arwen-wasm-vm/v1_3 v1.3.42-0.20220729115131-85ecca868e90 => github.com/ElrondNetwork/arwen-wasm-vm v1.3.42-0.20220729115131-85ecca868e90
-
```
Comment (iulianpascalau) [inline go.mod:56]: do we need the following replace for the v1.4 version?

**Review by iulianpascalau — COMMENTED** [review]
can we re-target this to rc/v1.4.0 branch? We are planning to phase out the "development" branch

File: go.mod (hunk 1 above)
Comment (camilbancioiu) [reply go.mod:56]: This PR will eventually obviate the need for the final `replace`. Work in progress,

Comment (camilbancioiu) [reply go.mod:56]: Done.

File: go.mod [hunk 2]
```diff
@@ -14,13 +11,16 @@ require (
 	github.com/ElrondNetwork/elrond-go-logger v1.0.7
 	github.com/ElrondNetwork/elrond-vm-common v1.3.15-0.20220803072537-08e036d340bc
 	github.com/ElrondNetwork/go-libp2p-pubsub v0.6.1-rc1
+	github.com/ElrondNetwork/wasm-vm v1.4.59-0.20220830135130-f4d8c00cf6dc
```
Comment (iulianpascalau) [inline go.mod:14]: some release tags would be great

**Review by iulianpascalau — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

File: process/factory/shard/vmContainerFactory.go [hunk 3]
```diff
@@ -5,12 +5,8 @@ import (
...
 	"sort"
 
-	arwen12 "github.com/ElrondNetwork/arwen-wasm-vm/v1_2/arwen"
-	arwenHost12 "github.com/ElrondNetwork/arwen-wasm-vm/v1_2/arwen/host"
-	arwen13 "github.com/ElrondNetwork/arwen-wasm-vm/v1_3/arwen"
-	arwenHost13 "github.com/ElrondNetwork/arwen-wasm-vm/v1_3/arwen/host"
-	arwen14 "github.com/ElrondNetwork/arwen-wasm-vm/v1_4/arwen"
-	arwenHost14 "github.com/ElrondNetwork/arwen-wasm-vm/v1_4/arwen/host"
+	arwen14 "github.com/ElrondNetwork/wasm-vm/arwen"
+	arwenHost14 "github.com/ElrondNetwork/wasm-vm/arwen/host"
```
Comment (sasurobert) [inline process/factory/shard/vmContainerFactory.go:9]: goimports

File: process/factory/shard/vmContainerFactory_test.go [hunk 4]
```diff
@@ -4,8 +4,7 @@ import (
 	"sync"
 	"testing"
 
-	ipcNodePart1p2 "github.com/ElrondNetwork/arwen-wasm-vm/v1_2/ipc/nodepart"
-	arwenConfig "github.com/ElrondNetwork/arwen-wasm-vm/v1_4/config"
+	arwenConfig "github.com/ElrondNetwork/wasm-vm/config"
 	"github.com/ElrondNetwork/elrond-go-core/core"
 	"github.com/ElrondNetwork/elrond-go-core/data"
```
Comment (sasurobert) [inline process/factory/shard/vmContainerFactory_test.go:9]: goimports

**Review by sasurobert — DISMISSED** [review]

File: process/factory/shard/vmContainerFactory.go (hunk 3 above)
Comment (camilbancioiu) [reply process/factory/shard/vmContainerFactory.go:9]: Fixing on a separate branch, for all files in the repo.

File: process/factory/shard/vmContainerFactory_test.go (hunk 4 above)
Comment (camilbancioiu) [reply process/factory/shard/vmContainerFactory_test.go:9]: Fixing on a separate branch, for all files in the repo.

**Review by sasurobert — DISMISSED** [review]

**Review by sstanculeanu — DISMISSED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by gabi-vuls — APPROVED** [review]
System test passed.

**Review by sstanculeanu — APPROVED** [review]

**Pull request merged**